app/code/VConnect/Blog/Service/PostUrlKeyChecker.php

```diff
@@ -0,0 +1,35 @@
+<?php
+declare(strict_types=1);
+
+namespace VConnect\Blog\Service;
+
+use VConnect\Blog\Model\ResourceModel\Post as PostResourceModel;
+
+class PostUrlKeyChecker
+{
+    /**
+     * PostUrlKeyChecker constructor.
+     * @param PostResourceModel $postResource
+     */
+    public function __construct(private PostResourceModel $postResource)
+    {}
+
+    /**
+     * @param string $urlKey
+     * @return string
+     */
+    public function checkUrlKey(string $urlKey): string
+    {
+        $connection = $this->postResource->getConnection();
+        $select = $connection->select()
+            ->from($connection->getTableName('blog_post_entity'), 'entity_id')
+            ->where('url_key = ?', $urlKey);
+
+        $result = $connection->fetchOne($select);
+        if (!$result) {
```

Comment:
i think logically here we can return url key or nothing. Here i prefer to make a return statement this like like this 
`return $connection->fetchOne($select) ?: null;`
And make a null checking in other methods.